src/main/java/org/hydev/mcpm/client/Downloader.java

```diff
@@ -17,16 +27,49 @@
     /** Number of simultaneous downloads */
     private int threads = 5;
 
+
     /**
      * Download one file from the internet to local storage through HTTP request
      *
      * @param url Remote file URL
      * @param to Local file path
      */
+
     public void downloadFile(String url, File to)
     {
         // TODO: Implement this
-        throw new UnsupportedOperationException("TODO");
+        // throw new UnsupportedOperationException("TODO");
+        try {
+            URL link = new URL(url);
+            HttpURLConnection http = (HttpURLConnection)link.openConnection();
+            long fileSize = (long)http.getContentLengthLong();
+
+
+            BufferedInputStream in = new BufferedInputStream(http.getInputStream());
+            FileOutputStream fileos = new FileOutputStream(to);
+            BufferedOutputStream bout = new BufferedOutputStream(fileos, 1024);
+            byte[] buffer = new byte[1024];
+            int read = 0;
+
+            var b = new ProgressBar(ProgressBarTheme.ASCII_THEME);
+            var all = new ArrayList<ProgressRow>();
+            var row = new ProgressRow(fileSize)
+                    .unit("MB")
```

Comment:
Is `MB` correct here? You increment by `1024` every time you read `1024` bytes. Do you mean `"bytes"`?